firefox-ios/Client/Frontend/Browser/SearchViewController.swift

```diff
@@ -1050,6 +1059,140 @@ private extension SearchViewController {
+        return Int32(filteredWords.count)
+    }
+
+    func numberOfSearchResults() -> Int {
+        return suggestions?.count ?? 0 + data.count + searchHighlights.count
+                + filteredOpenedTabs.count + firefoxSuggestions.count
+                + filteredRemoteClientTabs.count
+    }
+
+    // Comma separated list of result types in order.
+    func listResultTypes() -> String {
+        var resultTypes: [String] = []
+
+        // Iterate through each section
+        for sectionIndex in 0..<numberOfSections(in: tableView) {
+            let section = SearchListSection(rawValue: sectionIndex)!
+            // Iterate through each row in the section
+            for rowIndex in 0..<tableView(tableView, numberOfRowsInSection: sectionIndex) {
+                let indexPath = IndexPath(row: rowIndex, section: sectionIndex)
+                switch section {
+                case .searchSuggestions:
+                    resultTypes.append(TelemetryWrapper.EventExtraKey.UrlbarTelemetry.Results.searchSuggest.rawValue)
+                case .openedTabs:
+                    resultTypes.append(TelemetryWrapper.EventExtraKey.UrlbarTelemetry.Results.tab.rawValue)
+                case .remoteTabs:
+                    resultTypes.append(TelemetryWrapper.EventExtraKey.UrlbarTelemetry.Results.remoteTab.rawValue)
+                case .bookmarksAndHistory:
+                    // Check if the item is a bookmark or a history item
+                    if let site = data[indexPath.row] {
+                        resultTypes.append(site.bookmarked ?? false ?
+                                           TelemetryWrapper.EventExtraKey.UrlbarTelemetry.Results.bookmark.rawValue :
+                                            TelemetryWrapper.EventExtraKey.UrlbarTelemetry.Results.history.rawValue)
+                    }
+                case .searchHighlights:
+                    resultTypes.append(TelemetryWrapper.EventExtraKey.UrlbarTelemetry.Results.searchHistory.rawValue)
+                case .firefoxSuggestions:
+                    let firefoxSuggestion = firefoxSuggestions[indexPath.row]
+                    resultTypes.append(firefoxSuggestion.isSponsored ?
+                                       TelemetryWrapper.EventExtraKey.UrlbarTelemetry.Results.suggestSponsor.rawValue :
+                                        TelemetryWrapper.EventExtraKey.UrlbarTelemetry.Results.suggestNonSponsor.rawValue)
+                }
+            }
+        }
+
+        return resultTypes.joined(separator: ",")
+    }
+
+    // Comma separated list of result groups in order, groups may be
+    // repeated, since the list will match 1:1 the results list, so we
+    // Can link each result to a group
+    func listGroupTypes() -> String {
+        var groupTypes: [String] = []
+
+        for sectionIndex in 0..<numberOfSections(in: tableView) {
+            let section = SearchListSection(rawValue: sectionIndex)!
```

Comment:
do we want to avoid forced unwrap or handle it safely?

Reply:
this should be replaced with gurad or if let

Reply:
dropped the enum altogether